test/resque_test.rb

```diff
@@ -287,6 +287,12 @@
   end
 
   describe "stats" do
+    it "allows to set custom stat_data_store" do
+      dummy = DummyStatStore.new
+      Resque.stat_data_store = dummy
```

Comment:
You need to store the previous data_store and reset in an ensure block otherwise tests will fail.

Reply:
Done https://github.com/resque/resque/pull/1652/files#diff-8ce9de84c5e22ef89d9f1e236811c946R6